### PR TITLE
Dashboard: Fix template insertion

### LINKED ISSUE #12024: E2E Tests: Fix flakey Explore Templates test
## Feature Description

Test is failing looking for `Add title`

```
FAIL packages/e2e-tests/src/specs/dashboard/templates/useTemplate.js (30.295 s)
  ● Template › should be able to use existing template for new story

    TimeoutError: waiting for selector `input[placeholder="Add title"]` failed: timeout 3000ms exceeded

      65 |
      66 |     // Wait for title input to load before continuing.
    > 67 |     await page.waitForSelector('input[placeholder="Add title"]');
         |                ^
      68 |
      69 |     // Wait for skeleton thumbnails in the carousel to render which gives footer time to also render
      70 |     await page.waitForFunction(
```

### LINKED ISSUE #12345: Dashboard: Template insertion broken
## Bug Description

PR #12237 removed the elements registration from the dashboard app because it was deemed unused and thus redundant.

Turns out the elements are in fact needed when using the "Create story from template" functionality, because there we generate the AMP HTML output (via the `output` package) which in turn requires elements to be registered.

While we would have had an e2e test for this scenario that should have caught this right away, it has been disabled because of flakiness (see #12024)

The quick solution here is to revert #12237, the best solution would be to not use the `output` package at all since we don't actually need it.

## Expected Behaviour

Creating a new story from a template should be possible again.

## Steps to Reproduce

1. Go to "Explore Templates"
2. Try to create a new story from a template.

## Additional Context

- Plugin Version: 1.25.0
- WordPress Version: 6.0.2
- Operating System: macOS
- Browser: Chrome

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

#12237 accidentally broke the "Use template" functionality on the dashboard as it turns out the element types were in fact used after all. It wasn't caught because of a missing (disabled) e2e test.

But now no more!

## Summary

Instead of just reverting #12237, this PR removes usage of the `output` package from the dashboard to fix the template insertion functionality.

## Relevant Technical Choices

Fixes and re-enables the relevant e2e test.

## User-facing changes

No visual changes, just a fixed user flow

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Go to "Explore Templates"
2. Create a new story from a template
3. It should succeed

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12345 
Fixes #12024

### CODE REVIEW

File: includes/REST_API/Stories_Base_Controller.php [hunk 1]
```diff
@@ -127,11 +127,23 @@ protected function prepare_item_for_database( $request ) {
 		if ( ! empty( $schema['properties']['content'] ) ) {
 
 			// Ensure that content and story_data are updated together.
```
Comment (swissspidy) [inline includes/REST_API/Stories_Base_Controller.php:129]: Considering just removing this whole bit. Not sure if this has ever been useful.

Comment (miina) [reply includes/REST_API/Stories_Base_Controller.php:129]: Do you mean the check + throwing the error if not updated together to allow updating separately?

File: packages/wp-dashboard/src/api/story.js [hunk 2]
```diff
@@ -165,12 +163,7 @@ export const createStoryFromTemplate = async (config, template) => {
     title: '',
   };
 
-  const content = getStoryMarkup(story, pages, {
-    publisher: createdBy,
```
Comment (miina) [inline packages/wp-dashboard/src/api/story.js:169]: Does this mean that we're not using the `createdBy` field of templates anymore?

**Review by miina — COMMENTED** [review]
> But now no more!

Sounds like a radio ad 😄

File: includes/REST_API/Stories_Base_Controller.php (hunk 1 above)
Comment (swissspidy) [reply includes/REST_API/Stories_Base_Controller.php:129]: Yes

File: packages/wp-dashboard/src/api/story.js (hunk 2 above)
Comment (swissspidy) [reply packages/wp-dashboard/src/api/story.js:169]: We use it on the "Explore Templates" page to display who created the template.

**Review by timarney — APPROVED** [review]
Tested locally fixes the error currently on main

**Review by spacedmonkey — APPROVED** [review]

**Pull request merged**